### PR TITLE
More doc updates.

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

**Comment by nomis52** [comment]
We need a newer OLA .deb

File: user_manual/gen/dimmer.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<table class="doxtable"><tr><th>PID</th><th>Get</th><th>Set</th></tr>
+ <tr><td><a href="http://rdm.openlighting.org/pid/display?manufacturer=0&pid=48">STATUS_MESSAGES</a></td><td>Y</td><td></td></tr>
```
Comment (peternewman) [inline user_manual/gen/dimmer.html]: This should be & amp ; rather than just &.

File: user_manual/pid_gen/pid_gen.cpp [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+using ola::rdm::UID;
+using std::cerr;
+using std::cout;
+using std::endl;
+using std::setw;
+using std::string;
+using std::unique_ptr;
+using std::vector;
+
+const uint8_t TEST_UID[] = {0x7a, 0x70, 0, 0, 0, 1};
```
Comment (peternewman) [inline user_manual/pid_gen/pid_gen.cpp]: Not that it really matters, but any reason not to use the Ja-Rule non-MAC UID 7a70:fffffe00?

File: user_manual/pid_gen/pid_gen.cpp [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+const uint16_t MANUFACTURER_RANGE = 0x8000;
```
Comment (peternewman) [inline user_manual/pid_gen/pid_gen.cpp]: Why don't you link this from OLA?

File: user_manual/pid_gen/pid_gen.cpp [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+string BuildLink(uint16_t pid) {
+  std::stringstream str;
+  str << "http://rdm.openlighting.org/pid/display?manufacturer=";
+  if (pid < MANUFACTURER_RANGE) {
+    str << 0;
+  } else {
+    str << ola::OPEN_LIGHTING_ESTA_CODE;
+  }
+  str << "&pid=" << pid;
```
Comment (peternewman) [inline user_manual/pid_gen/pid_gen.cpp]: Make this & amp ;

File: user_manual/pid_gen/pid_gen.cpp [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+  std::sort(rows.begin(), rows.end());
+  OutputTable(model.name, rows);
+}
+
+int main(int argc, char* argv[]) {
+  ola::AppInit(
+      &argc,
+      argv,
+      "[options]",
+      "generate Markdown tables for each models supported parameters");
```
Comment (peternewman) [inline user_manual/pid_gen/pid_gen.cpp]: SPaG model's

**Comment by peternewman** [comment]
Can you add this to Travis cpplint checking. We also need to ponder on the Travis Doxygen/general checking of this code.

**Comment by peternewman** [comment]
The GCC error isn't down to the age of OLA, just better detection their end I suspect:
https://travis-ci.org/OpenLightingProject/ja-rule/jobs/112464786
libtool: compile:  gcc-5 -DHAVE_CONFIG_H -I. -I boardcfg -I tests/system_config -I tests/harmony/include -I firmware/src -I common -I Bootloader/firmware/src -Wall -Wformat -W -Werror -g -O2 -pthread -MT firmware/src/firmware_src_libmovinglightmodel_la-moving_light.lo -MD -MP -MF firmware/src/.deps/firmware_src_libmovinglightmodel_la-moving_light.Tpo -c firmware/src/moving_light.c  -fPIC -DPIC -o firmware/src/.libs/firmware_src_libmovinglightmodel_la-moving_light.o
firmware/src/moving_light.c: In function ‘MovingLightModel_SetPowerState’:
firmware/src/moving_light.c:367:21: error: comparison is always false due to limited range of data type [-Werror=type-limits]
   if (param_data[0] > POWER_STATE_NORMAL) {
                     ^
cc1: all warnings being treated as errors
make: *** [firmware/src/firmware_src_libmovinglightmodel_la-moving_light.lo] Error 1

File: user_manual/gen/dimmer.html (hunk 1 above)
Comment (nomis52) [inline user_manual/gen/dimmer.html]: Done

File: user_manual/pid_gen/pid_gen.cpp (hunk 2 above)
Comment (nomis52) [inline user_manual/pid_gen/pid_gen.cpp]: I switched to the development range.

File: user_manual/pid_gen/pid_gen.cpp (hunk 3 above)
Comment (nomis52) [inline user_manual/pid_gen/pid_gen.cpp]: I couldn't actually find it in OLA :)

File: user_manual/pid_gen/pid_gen.cpp (hunk 4 above)
Comment (nomis52) [inline user_manual/pid_gen/pid_gen.cpp]: Done

File: user_manual/pid_gen/pid_gen.cpp (hunk 5 above)
Comment (nomis52) [inline user_manual/pid_gen/pid_gen.cpp]: Done

**Comment by nomis52** [comment]
I'm going to merge this so we get updated docs and then we'll figure out the .debs

**Comment by peternewman** [comment]
Your Doxygen HTML includes weren't working @nomis52 . I assume you manually ran the gen script? We should add it to the buildbot job.

**Comment by nomis52** [comment]
buildbot built a change at a time, once it got to the right one it worked.

**Comment by peternewman** [comment]
Ah, it looked like it only built with this force build:
https://buildbot.openlighting.org/builders/doxygen-manual-generator-ja-rule/builds/15

Either way, I don't see "make doc-gen" being run by the buildbot; surely we should do that there, so the manual stays up to date?

**Comment by nomis52** [comment]
It worked for  #13 as well:

https://buildbot.openlighting.org/builders/doxygen-manual-generator-ja-rule/builds/13

My mistake was triggering a build for 928504398 after the build for a59e7670ab2.

You can add make doc-gen to the buildbot if you like. Or we can just re-run it when the code changes.

**Pull request merged**